python_scripts/repeatable/strip_whitespace.py

```diff
@@ -0,0 +1,142 @@
+        field_2 (str): the key of the nested field to strip whitespace from. Will be blank if the
+        targeted field is not nested in a subrecord.
+    """
+    updated_object = {}
+    if field_1 in json_object:
+        object_1 = json_object[field_1]
+        if field_2:
+            for i, item in list(enumerate(object_1)):
+                json_object[field_1][i][field_2] = item[field_2].strip()
+                updated_object = json_object
+                return updated_object
+        else:
+            json_object[field_1] = object_1.strip()
+            updated_object = json_object
+            return updated_object
+    else:
+        record_error(f'Field `{field_1}` does not exist in JSON data: ',
+                     json_object)
+
+def main(whitespace_csv, dry_run):
+    """
+    Runs the functions of the script, fetching, digging into, and stripping whitespace from a
+    specified record type and field, printing error messages if they occur.
+
+    Takes a csv input of existing ASpace records with, at minimum:
+        - id - The id of the record to be updated (ex: a resource id or digital object id).
+        - field_1 - The name of the field with offending whitespace.  This is the field name as represented
+        in the JSON returned from ArchivesSpace (ex: 'title').  If the field is nested deeper in the record,
+        this will be the name of the parent (for example, if wishing to strip whitespace from a digital 
+        object's nested file_uri, field_1 would be 'file_versions').
+        - field_2 - For nested fields, the name of the field with offending whitespace.  This is the field name
+        as represented in the JSON returned from ArchivesSpace (ex: 'file_uri').  This will be blank if the
+        field in field_1 is the field targeted for update.
+        - repo_id - The repo_id of the record to be updated
+
+    Args:
+        whitespace_csv (str): filepath for the csv
+        dry_run (bool): run as non-destructive dry run?  True if `--dry-run` provided as an argument
+    """
+
+    aspace = ASpace(
+        baseurl=os.getenv('as_api'),
+        username=os.getenv('as_un'),
+        password=os.getenv('as_pw')
+    )
+    csv_dict = read_csv(whitespace_csv)
+    for obj in csv_dict:
+        repo = aspace.repositories(obj['repo_id'])
+        existing_object = func_dict[args.type](repo)(obj['id'])
+        if isinstance(existing_object, dict) and 'error' in existing_object:
+            record_error(f'Unable to retrieve object with provided URI: '
+                         f'repositories/{obj["repo_id"]}/{args.type}/{obj["id"]}',
+                         existing_object)
+            return None
+        elif existing_object is not None:
+            json_object = existing_object.json()
+            data = strip_whitespace(json_object, obj['field_1'], obj['field_2'])
+            if not dry_run:
+                update_message = aspace.client.post(data['uri'], json=data).json()
```

Comment:
Out of curiosity, did your IDE throw a warning that aspace.client does not contain the method .post? Mine does that when I'm importing the ASnakeClient.

Reply:
No, because ASnake client does have a post: https://archivesspace-labs.github.io/ArchivesSnake/html/index.html#asnake.client.ASnakeClient